### PR TITLE
enable paddle.static.gradients for supporting primitive autograd

### PR DESCRIPTION
**Pull request by cxxly**
### PR types

New features
### PR changes

APIs
### Describe

enable paddle.static.gradients for supporting primitive autograd.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/autograd/test_primops.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
     transpose, split, concat, reduce, matmul, slice_select, slice_assign,
     gather, scatter_add, fill_const)
 from paddle.incubate.autograd.primx import Transform, topo_path, orig2prim, prim2orig, _gradients
-from paddle.incubate.autograd.utils import enable_prim
+from paddle.incubate.autograd.utils import enable_prim, disable_prim, prim_enabled
 
 
 def prog1(x, y):
```
Comment (0x45f) [inline python/paddle/fluid/tests/unittests/autograd/test_primops.py]: `prog1 `和`prog2 `两个函数是不是可以删除

File: python/paddle/fluid/tests/unittests/autograd/test_transform.py [hunk 2]
```diff
@@ -310,65 +310,186 @@ def init_data(self):
...
+        y = np.random.rand(5, 5)
+        disable_prim()
+        origs = matmul_second_order(x, y)
+        enable_prim()
+        prims = matmul_second_order(x, y)
+        disable_prim()
+        for orig, prim in zip(origs, prims):
+            np.testing.assert_allclose(orig, prim)
+
+    def test_three_order(self):
```
Comment (levi131) [inline python/paddle/fluid/tests/unittests/autograd/test_transform.py]: test_third_order

File: python/paddle/fluid/tests/unittests/autograd/test_primops.py (hunk 1 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/autograd/test_primops.py]: updated

File: python/paddle/fluid/tests/unittests/autograd/test_transform.py (hunk 2 above)
Comment (cxxly) [reply python/paddle/fluid/tests/unittests/autograd/test_transform.py]: updated

**Comment by levi131** [comment]
LGTM

**Pull request merged**